python/hail/methods/impex.py

```diff
@@ -615,16 +615,122 @@ def import_table(paths, key=[], min_partitions=None, impute=False, no_header=Fal
            contig_recoding=nullable(dictof(strlike, strlike)),
            drop_chr0=bool)
-def import_plink(bed, bim, fam, min_partitions=None, delimiter='\\\\s+',
-                 missing='NA', quant_pheno=False, a2_reference=True, reference_genome=None,
-                 contig_recoding={'23': 'X', '24': 'Y', '25': 'X', '26': 'MT'}, drop_chr0=False):
+def import_plink(bed, bim, fam,
+                 min_partitions=None,
+                 delimiter='\\\\s+',
+                 missing='NA',
+                 quant_pheno=False,
+                 a2_reference=True,
+                 reference_genome=None,
+                 contig_recoding={'23': 'X',
+                                  '24': 'Y',
+                                  '25': 'X',
+                                  '26': 'MT'},
+                 drop_chr0=False):
+    """Import PLINK binary file (BED, BIM, FAM) as a :class:`.MatrixTable`.
+
+    Examples
+    --------
+
+    Import data from a PLINK binary file:
+
+    >>> ds = methods.import_plink(bed="data/test.bed",
+    ...                            bim="data/test.bim",
+    ...                            fam="data/test.fam")
+
+    Notes
+    -----
+
+    Only binary SNP-major mode files can be read into Hail. To convert your
+    file from individual-major mode to SNP-major mode, use PLINK to read in
+    your fileset and use the ``--make-bed`` option.
+
+    The centiMorgan position is not currently used in Hail (Column 3 in BIM
+    file).
+
+    The ID (``s``) used by Hail is the individual ID (column 2 in FAM file).
+
+    .. warning::
+
+        No duplicate individual IDs are allowed.
+
+    The produced :class:`.MatrixTable` has one column-indexd field:
+
+     - **rsid** (*String*) -- Column 2 in the BIM file.
+
+    and six row-indexd fields:
+
+     - **famID** (*String*) -- Column 1 in the FAM file. Set to missing if
+       ID equals "0".
+     - **patID** (*String*) -- Column 3 in the FAM file. Set to missing if
+       ID equals "0".
+     - **matID** (*String*) -- Column 4 in the FAM file. Set to missing if
+       ID equals "0".
+     - **isFemale** (*String*) -- Column 5 in the FAM file. Set to missing
+       if value equals "-9", "0", or "N/A". Set to true if value equals "2".
+       Set to false if value equals "1".
+     - **isCase** (*String*) -- Column 6 in the FAM file. Only present if
+       ``quantpheno`` equals False. Set to missing if value equals "-9", "0",
+       "N/A", or the value specified by ``missing``. Set to true if value
+       equals "2". Set to false if value equals "1".
+     - **qPheno** (*String*) -- Column 6 in the FAM file. Only present if
+       ``quantpheno`` equals True. Set to missing if value equals ``missing``.
+
+    Parameters
+    ----------
+    bed : :obj:`str`
+        PLINK BED file.
+
+    bim : :obj:`str`
+        PLINK BIM file.
+
+    fam : :obj:`str`
+        PLINK FAM file.
+
+    min_partitions : :obj:`int` or :obj:`None`
```

Comment:
Tim prefers:
```
:obj:`int`, optional
```

Reply:
Oh, I must have missed that discussion. We should probably make sure that's clear somewhere. The NumPy style (which we're otherwise adopting) uses optional to mean "has a default value".

Reply:
does is use "or None" for things which can be none?

Reply:
this has a default value, just not one known statically

Reply:
I'm going off of https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt. In the parameters section, it says to use optional for keyword args which don't have to be provided, because they have defaults. I couldn't find any examples of a NumPy argument which can be `None`.